src/cdk/utils/create-lambda-integration.ts

```diff
@@ -52,6 +63,10 @@ export function createLambdaIntegration(
       }
     ),
     {
+      authorizationType: authenticationRequired
+        ? AuthorizationType.CUSTOM
+        : AuthorizationType.NONE,
+      authorizer: authenticationRequired ? undefined : noneAuthorizer,
```

Comment:
Pass authorizer into this function, and throw if authenticationRequired is true but no authorizer is passed